_episodes/09-github.md

```diff
@@ -93,18 +89,184 @@ We can check that the command has worked by running `git remote -v`:
+
+You can think of the public key as a padlock, and only you have the key (the private key) to open it. You use the public key where you want a secure method of communication, such as your GitHub account.  You give this padlock, or public key, to GitHub and say “lock the communications to my account with this so that only computers that have my private key can unlock communications and send git commands as my GitHub account.”  
+
+What we will do now is the minimum required to set up the SSH keys and add the public key to a GitHub account.
+
+> ## Advanced SSH
+> A supplemental episode in this lesson discusses SSH and key pairs in more depth and detail. 
+{: .callout}
+
+The first thing we are going to do is check if this has already been done on the computer you’re on.  Because generally speaking, this setup only needs to happen once and then you can forget about it. 
+
+> ## Keeping your keys secure
+> You shouldn't really forget about your SSH keys, since they keep your account secure. It’s good 
+>  practice to audit your secure shell keys every so often. Especially if you are using multiple 
+>  computers to access your account.
+{: .callout}
+
+We will run the list command to check what key pairs already exist on your computer.
+
+~~~
+ls -al ~/.ssh
+~~~
+{: .language-bash}
+
+Your output is going to look a little different depending on whether or not SSH has ever been set up on the computer you are using. 
+
+Dracula has not set up SSH on his computer, so his output is 
+
+~~~
+ls: cannot access '/c/Users/Vlad Dracula/.ssh': No such file or directory
+~~~
+{: .output}
+
+If SSH has been set up on the computer you're using, the public and private key pairs will be listed. The file names are either `id_ed25519`/`id_ed25519.pub` or `id_rsa`/`id_rsa.pub` depending on how the key pairs were set up.  
+Since they don’t exist on Dracula’s computer, he uses this command to create them. 
+
+### Create an SSH key pair
+To create an SSH key pair Vlad uses this command, where the `-t` option specifies which type of algorithm to use and `-C` attaches a comment to the key (here, Vlad's email):  
+
+~~~
+$ ssh-keygen -t ed25519 -C "vlad@tran.sylvan.ia"
+~~~
+{: .language-bash}
+
+If you are using a legacy system that doesn't support the Ed25519 algorithm, use:
+`$ ssh-keygen -t rsa -b 4096 -C "your_email@example.com"`
+
+~~~
+Generating public/private ed25519 key pair.
+Enter file in which to save the key (/c/Users/Vlad Dracula/.ssh/id_ed25519):
+~~~
+{: .output}
+
+We want to use the default file, so just press <kbd>Enter</kbd>.
+
+~~~
+Created directory '/c/Users/Vlad Dracula/.ssh'.
+Enter passphrase (empty for no passphrase):
+~~~
+{: .output}
+
+Now, it is prompting Dracula for a passphrase.  Since he is using his lab’s laptop that other people sometimes have access to, he wants to create a passphrase.  Be sure to use something memorable or save your passphrase somewhere, as there is no "reset my password" option. 
```

Comment:
I'd put a note here that the unix shell won't show anything when you type in your password, not even `****`.  This is a common issue for learners.